### PR TITLE
[Modules] Upgraded AKS API version to latest non-review

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Upgraded AKS API version to latest non-review

| Pipeline |
| - |
| [ContainerService: ManagedClusters](https://github.com/Azure/ResourceModules/actions/workflows/ms.containerservice.managedclusters.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.ContainerService/managedClusters/agentPools/deploy.bicep [hunk 1]
```diff
@@ -187,11 +187,11 @@ resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (ena
   }
 }
 
-resource managedCluster 'Microsoft.ContainerService/managedClusters@2021-08-01' existing = {
+resource managedCluster 'Microsoft.ContainerService/managedClusters@2022-06-01' existing = {
```
Comment (eriqua) [inline modules/Microsoft.ContainerService/managedClusters/agentPools/deploy.bicep:190]: For consistency I'd suggest to also update to the same version references to this resource type in other modules (e.g. the ones in modules\Microsoft.KubernetesConfiguration)

Comment (MrMCake) [reply modules/Microsoft.ContainerService/managedClusters/agentPools/deploy.bicep:190]: Can't say I agree - but will do to get this PR in ;)

Comment (MrMCake) [reply modules/Microsoft.ContainerService/managedClusters/agentPools/deploy.bicep:190]: done

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**